### PR TITLE
Repo cleanup

### PR DESCRIPTION
**Pull request by grandizzy**
- Ajna rewards contract cleanup
  - stake / unstake functions
  - overrides, inheritdoc
  - rename some state vars / mappings
  - AjnaRewards -> RewardsManager refactor
  - moved in proper (for now) base dir with similar interfaces like other contracts (actions, state, errors, events, derrived state)

- Position manager contract cleanup
  - remove unused events
  - change price occurences to bucket to be inline with other contracts (e.g. bucket index instead price index)
  - comments, inheritdoc
  - improved naming (lps -> positionLPs)
  - add position manager state interface and error interface

- Libraries cleanup: do not repeat same comments as in Pool errors/events just point to them, more comments
- Add doc with function description, improved functions comments
- use relative paths in imports

### CODE REVIEW

**Comment by MikeHathaway** [comment]
Fine to leave the relative path imports as is while still working on the repo, but probably want to change to absolute path before going live

File: src/base/RevertsHelper.sol
Comment (MikeHathaway) [inline src/base/RevertsHelper.sol]: should be `IPoolErrors`

File: src/base/interfaces/position/IPositionManagerErrors.sol
Comment (MikeHathaway) [inline src/base/interfaces/position/IPositionManagerErrors.sol]: Can be removed, error no longer used.

File: src/base/interfaces/position/IPositionManagerOwnerActions.sol
Comment (MikeHathaway) [inline src/base/interfaces/position/IPositionManagerOwnerActions.sol]: Should we remove this `TODO:`, and leave it up to the client to determine how many buckets can be memorialized? Viability of any cap would be very different per chain

File: src/base/interfaces/rewards/IRewardsManagerDerivedState.sol
Comment (MikeHathaway) [inline src/base/interfaces/rewards/IRewardsManagerDerivedState.sol:8]: Should we just move the `...DerivedState` interfaces into `...State`? The files aren't too big, and it's less clear than the other modules.

File: src/base/interfaces/position/IPositionManagerOwnerActions.sol
Comment (grandizzy) [reply src/base/interfaces/position/IPositionManagerOwnerActions.sol]: agree with this, how does
`*  @dev    The NFT must have already been created, and the number of buckets to be memorialized at a time determined by function caller.`
sound to you?

File: src/base/interfaces/rewards/IRewardsManagerEvents.sol
Comment (MikeHathaway) [inline src/base/interfaces/rewards/IRewardsManagerEvents.sol]: Should we rename these to `Stake` and `Unstake` to match the new function sig?

Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerEvents.sol]: I tried that but conflicts with `State` struct. Can rename existing `Stake` struct to `StakeState` or `StakeInfo` and rename events, wdyt?

File: src/base/interfaces/rewards/IRewardsManagerDerivedState.sol
Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerDerivedState.sol:8]: it makes sense for `getStakeInfo` to be moved into `State` interface, but not sure about `calculateRewards`, I'd keep it in derived as it requires extra computations...

File: src/base/interfaces/rewards/IRewardsManagerState.sol
Comment (MikeHathaway) [inline src/base/interfaces/rewards/IRewardsManagerState.sol:64]: Could use comments on this, including decimal precision

File: src/erc20/ERC20PoolFactory.sol
Comment (MikeHathaway) [inline src/erc20/ERC20PoolFactory.sol:9]: Not very important, but should we set a consistent approach for dealing with import ordering / alignment? Thought previous of interfaces with interfaces should hold here

File: src/libraries/Buckets.sol
Comment (MikeHathaway) [inline src/libraries/Buckets.sol]: should be `IPoolError`

File: src/erc20/ERC20PoolFactory.sol
Comment (grandizzy) [reply src/erc20/ERC20PoolFactory.sol:9]: agree we should be consistent but not sure I understand the order you propose, can you put an example?

File: src/base/interfaces/position/IPositionManagerOwnerActions.sol
Comment (MikeHathaway) [reply src/base/interfaces/position/IPositionManagerOwnerActions.sol]: Sounds good!

File: src/base/interfaces/rewards/IRewardsManagerEvents.sol
Comment (MikeHathaway) [reply src/base/interfaces/rewards/IRewardsManagerEvents.sol]: Ah right. Yeah maybe rename the struct to `StakeInfo` as the function and event symmetry would match the rest of the codebase

**Comment by MikeHathaway** [comment]
Could also be helpful to have the precision in the struct definitions like [WAD], or [RAY]

File: src/base/interfaces/rewards/IRewardsManagerDerivedState.sol
Comment (MikeHathaway) [reply src/base/interfaces/rewards/IRewardsManagerDerivedState.sol:8]: I guess i'm more worried about people being confused by having to navigate between too many different files, so even if some information would be lost (like the fact that it's derived for external viewing) that might still net out to be easier for readers

File: src/libraries/external/Auctions.sol
Comment (MikeHathaway) [inline src/libraries/external/Auctions.sol:88]: wouldn't it be cleaner to leave these spaces in?

**Review by MikeHathaway — COMMENTED** [review]
Generally looks great! Just had a few styling comments, but otherwise a big improvement

File: src/libraries/external/Auctions.sol
Comment (grandizzy) [reply src/libraries/external/Auctions.sol:88]: I find nicer without as code is compacted :)

File: src/base/interfaces/rewards/IRewardsManagerDerivedState.sol
Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerDerivedState.sol:8]: this way we could compact everything in a single interface but will be different than what we have for pools for example and I wanted to keep the same...

File: src/erc20/ERC20PoolFactory.sol
Comment (MikeHathaway) [reply src/erc20/ERC20PoolFactory.sol:9]: How about: external <space> base <space> interfaces, with compacting in each group. Think that's similar to what we've done elsewhere

File: src/base/RevertsHelper.sol
Comment (grandizzy) [reply src/base/RevertsHelper.sol]: fixed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/base/interfaces/position/IPositionManagerErrors.sol
Comment (grandizzy) [reply src/base/interfaces/position/IPositionManagerErrors.sol]: fixed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/base/interfaces/position/IPositionManagerOwnerActions.sol
Comment (grandizzy) [reply src/base/interfaces/position/IPositionManagerOwnerActions.sol]: fixed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/base/interfaces/rewards/IRewardsManagerDerivedState.sol
Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerDerivedState.sol:8]: fixed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/base/interfaces/rewards/IRewardsManagerEvents.sol
Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerEvents.sol]: changed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/base/interfaces/rewards/IRewardsManagerState.sol
Comment (grandizzy) [reply src/base/interfaces/rewards/IRewardsManagerState.sol:64]: added for all structs with https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

File: src/libraries/Buckets.sol
Comment (grandizzy) [reply src/libraries/Buckets.sol]: fixed in https://github.com/ajna-finance/contracts/pull/505/commits/9c9e00f022118c2f23339de06bf0e5cc2e59d429

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/base/RevertsHelper.sol
Comment (EdNoepel) [inline src/base/RevertsHelper.sol:19]: hope the auditors are happy with that

File: src/erc20/ERC20Pool.sol
Comment (EdNoepel) [inline src/erc20/ERC20Pool.sol:109]: Why do we list reverts on `repayDebt` but not here?  This can revert on `BorrowerNotSender`, `LimitIndexReached`, `BorrowerUnderCollateralized`, `AmountLTMinDebt`, and `DustAmountNotExceeded`.

File: src/libraries/external/Auctions.sol
Comment (EdNoepel) [inline src/libraries/external/Auctions.sol:984]: Did this move somewhere else?  If not, can we now remove `DSTestPlus._assertArbTakeInsufficentCollateralRevert` and `_assertDepositTakeInsufficentCollateralRevert`?  How are tests expecting this revert still passing?

**Review by EdNoepel — APPROVED** [review]
Some comments inline.  Please review before merging.

File: src/libraries/external/Auctions.sol
Comment (grandizzy) [reply src/libraries/external/Auctions.sol:984]: yeah, I moved it before `_takeBucket` function https://github.com/ajna-finance/contracts/blob/repo-cleanup/src/libraries/external/Auctions.sol#L480 to be inline with `take`: https://github.com/ajna-finance/contracts/blob/repo-cleanup/src/libraries/external/Auctions.sol#L545 Tests are not affected

File: src/erc20/ERC20Pool.sol
Comment (grandizzy) [reply src/erc20/ERC20Pool.sol:109]: per Ian's request I listed only the reverts that happens in current function and not in functions that are called by it. So this one within Pool contract doesn't revert but the one from external BorrowerActions library does, so it is listed there: https://github.com/ajna-finance/contracts/blob/repo-cleanup/src/libraries/external/BorrowerActions.sol#L226

File: src/base/RevertsHelper.sol
Comment (grandizzy) [reply src/base/RevertsHelper.sol:19]: That's the style used by AAVE too so assume it's a common practice...

**Pull request merged**
